### ISSUE TITLE
Map tiles not displayed on when visiting localhost:8080

### ISSUE THREAD

**Issue by chris442** [Overv/openstreetmap-tile-server#128]
I downloaded the https://download.geofabrik.de/europe/britain-and-ireland-latest.osm.pbf map files and have created a Docker container following the guide in the Readme.

The output looks ok:

```
PS C:\Development\Uni> docker run -v C:\Development\Uni\MScProject\mapdata\britain-and-ireland-latest.osm.pbf:/data.osm.pbf -v openstreetmap-data:/var/lib/postgres/12/main overv/openstreetmap-tile-server import
Unable to find image 'overv/openstreetmap-tile-server:latest' locally
latest: Pulling from overv/openstreetmap-tile-server
23884877105a: Pull complete                                                                                             bc38caa0f5b9: Pull complete                                                                                             2910811b6c42: Pull complete                                                                                             36505266dcc6: Pull complete                                                                                             a7fd375a0556: Pull complete                                                                                             5d2376655a04: Pull complete                                                                                             6fc13a9ef95c: Pull complete                                                                                             661912a83af9: Pull complete                                                                                             383730879a6f: Pull complete                                                                                             c5fa28530964: Pull complete                                                                                             a53b7d95bf36: Pull complete                                                                                             b137e7d935a4: Pull complete                                                                                             f5b52a28bf23: Pull complete                                                                                             f107841f4b7c: Pull complete                                                                                             a5984c0374a0: Pull complete                                                                                             d4d8fd8f9606: Pull complete                                                                                             fd81e24bbbae: Pull complete                                                                                             94d05089bf69: Pull complete                                                                                             92a6106c2592: Pull complete                                                                                             d71c28cd6281: Pull complete                                                                                             7c245b2ff8ba: Pull complete                                                                                             84f4c9994ded: Pull complete                                                                                             9cdb5e74aae1: Pull complete                                                                                             87873f280ca1: Pull complete                                                                                             f15012519b5e: Pull complete                                                                                             Digest: sha256:cacf034aa8fccf4f4c1d2108bbc8946c4b67f7390dabffcc677f81e78daa3efe
Status: Downloaded newer image for overv/openstreetmap-tile-server:latest
+ '[' 1 -ne 1 ']'
+ '[' import = import ']'
+ chown postgres:postgres -R /var/lib/postgresql
+ '[' '!' -f /var/lib/postgresql/12/main/PG_VERSION ']'
+ createPostgresConfig
+ cp /etc/postgresql/12/main/postgresql.custom.conf.tmpl /etc/postgresql/12/main/conf.d/postgresql.custom.conf
+ sudo -u postgres echo 'autovacuum = on'
+ cat /etc/postgresql/12/main/conf.d/postgresql.custom.conf
# Suggested minimal settings from
# https://ircama.github.io/osm-carto-tutorials/tile-server-ubuntu/

shared_buffers = 128MB
min_wal_size = 1GB
max_wal_size = 2GB
maintenance_work_mem = 256MB

# Suggested settings from
# https://github.com/openstreetmap/chef/blob/master/roles/tile.rb#L38-L45

max_connections = 250
temp_buffers = 32MB
work_mem = 128MB
wal_buffers = 1024kB
wal_writer_delay = 500ms
commit_delay = 10000
# checkpoint_segments = 60 # unrecognized in psql 10.7.1
max_wal_size = 2880MB
random_page_cost = 1.1
track_activity_query_size = 16384
autovacuum_vacuum_scale_factor = 0.05
autovacuum_analyze_scale_factor = 0.02

listen_addresses = '*'
autovacuum = on
+ service postgresql start
 * Starting PostgreSQL 12 database server
   ...done.
+ sudo -u postgres createuser renderer
+ sudo -u postgres createdb -E UTF8 -O renderer gis
+ sudo -u postgres psql -d gis -c 'CREATE EXTENSION postgis;'
CREATE EXTENSION
+ sudo -u postgres psql -d gis -c 'CREATE EXTENSION hstore;'
CREATE EXTENSION
+ sudo -u postgres psql -d gis -c 'ALTER TABLE geometry_columns OWNER TO renderer;'
ALTER TABLE
+ sudo -u postgres psql -d gis -c 'ALTER TABLE spatial_ref_sys OWNER TO renderer;'
ALTER TABLE
+ setPostgresPassword
+ sudo -u postgres psql -c 'ALTER USER renderer PASSWORD '\''renderer'\'''
ALTER ROLE
+ '[' '!' -f /data.osm.pbf ']'
+ '[' -n '' ']'
+ '[' disabled = enabled ']'
+ '[' -f /data.poly ']'
+ sudo -u renderer osm2pgsql -d gis --create --slim -G --hstore --tag-transform-script /home/renderer/src/openstreetmap-carto/openstreetmap-carto.lua --number-processes 4 -S /home/renderer/src/openstreetmap-carto/openstreetmap-carto.style /data.osm.pbf
osm2pgsql version 1.2.0

Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Allocating memory for sparse node cache
Sharing dense sparse
Node-cache: cache=800MB, maxblocks=12800*65536, allocation method=11
Mid: pgsql, cache=800
Setting up table: planet_osm_nodes
Setting up table: planet_osm_ways
Setting up table: planet_osm_rels
Using lua based tag processing pipeline with script /home/renderer/src/openstreetmap-carto/openstreetmap-carto.lua
Using projection SRS 3857 (Spherical Mercator)
Setting up table: planet_osm_point
Setting up table: planet_osm_line
Setting up table: planet_osm_polygon
Setting up table: planet_osm_roads

Reading in file: /data.osm.pbf
Using PBF parser.
Processing: Node(146097k 529.3k/s) Way(19434k 7.21k/s) Relation(280992 373.2/s)  parse time: 3725s
Node stats: total(146097210), max(7447293813) in 276s
Way stats: total(19434786), max(796307079) in 2695s
Relation stats: total(280992), max(11035562) in 753s
Sorting data and creating indexes for planet_osm_point
Sorting data and creating indexes for planet_osm_polygon
Sorting data and creating indexes for planet_osm_line
Sorting data and creating indexes for planet_osm_roads
Using native order for clustering
Using native order for clustering
Using native order for clustering
Using native order for clustering
Copying planet_osm_point to cluster by geometry finished
Creating geometry index on planet_osm_point
Copying planet_osm_roads to cluster by geometry finished
Creating geometry index on planet_osm_roads
Creating osm_id index on planet_osm_roads
Creating indexes on planet_osm_roads finished
All indexes on planet_osm_roads created in 108s
Completed planet_osm_roads
Stopping table: planet_osm_nodes
Stopped table: planet_osm_nodes in 0s
Stopping table: planet_osm_ways
Building index on table: planet_osm_ways
Creating osm_id index on planet_osm_point
Creating indexes on planet_osm_point finished
All indexes on planet_osm_point created in 201s
Completed planet_osm_point
Stopping table: planet_osm_rels
Building index on table: planet_osm_rels
Stopped table: planet_osm_rels in 17s
Copying planet_osm_line to cluster by geometry finished
Creating geometry index on planet_osm_line
Creating osm_id index on planet_osm_line
Creating indexes on planet_osm_line finished
All indexes on planet_osm_line created in 511s
Completed planet_osm_line
Copying planet_osm_polygon to cluster by geometry finished
Creating geometry index on planet_osm_polygon
Creating osm_id index on planet_osm_polygon
Creating indexes on planet_osm_polygon finished
All indexes on planet_osm_polygon created in 789s
Completed planet_osm_polygon
Stopped table: planet_osm_ways in 1157s

Osm2pgsql took 4991s overall
node cache: stored: 52860319(36.18%), storage efficiency: 50.41% (dense blocks: 229, sparse nodes: 51490817), hit rate: 37.50%
+ sudo -u postgres psql -d gis -f indexes.sql
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
+ touch /var/lib/mod_tile/planet-import-complete
+ service postgresql stop
 * Stopping PostgreSQL 12 database server
   ...done.
+ exit 0
```

And so I run the container, but when I visit localhost:8080 the page appears to load ok with the page title "Custome Tile Server" listed and zoom in and out buttons for the map and Leaflet copyright details in the bottom right corner except the all important map.

When I view the network tab I can see there are 404 errors so I viewed the docker logs to see what is being reported on the server:
```

renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(3), y(3), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/3/3) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(4), y(3), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/4/3) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(3), y(4), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/3/4) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming connection, fd 7, number 2
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(3), y(2), mime(image/png), options()
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(7) xml(ajt), z(3), x(2), y(3), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/3/2) with protocol version 2 to fd 6
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/2/3) with protocol version 2 to fd 7
renderd[146]: DEBUG: Got incoming connection, fd 8, number 3
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 2 left
renderd[146]: DEBUG: Connection 0, fd 7 closed, now 1 left
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(8) xml(ajt), z(3), x(4), y(2), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 6, number 2
renderd[146]: DEBUG: Got incoming connection, fd 7, number 3
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(5), y(3), mime(image/png), options()
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(7) xml(ajt), z(3), x(2), y(4), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 9, number 4
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(9) xml(ajt), z(3), x(5), y(4), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 10, number 5
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(10) xml(ajt), z(3), x(3), y(5), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 11, number 6
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(11) xml(ajt), z(3), x(4), y(5), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/4/2) with protocol version 2 to fd 8
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/4/5) with protocol version 2 to fd 11
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/3/5) with protocol version 2 to fd 10
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/5/4) with protocol version 2 to fd 9
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/2/4) with protocol version 2 to fd 7
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/5/3) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 1, fd 6 closed, now 5 left
renderd[146]: DEBUG: Connection 2, fd 9 closed, now 4 left
renderd[146]: DEBUG: Connection 2, fd 10 closed, now 3 left
renderd[146]: DEBUG: Connection 1, fd 7 closed, now 2 left
renderd[146]: DEBUG: Connection 1, fd 11 closed, now 1 left
renderd[146]: DEBUG: Connection 0, fd 8 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(2), y(2), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/2/2) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(5), y(2), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 7, number 2
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(7) xml(ajt), z(3), x(2), y(5), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 8, number 3
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(8) xml(ajt), z(3), x(1), y(3), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 9, number 4
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/5/2) with protocol version 2 to fd 6
renderd[146]: DEBUG: Got command RenderPrio fd(9) xml(ajt), z(3), x(5), y(5), mime(image/png), options()
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/1/3) with protocol version 2 to fd 8
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 3 left
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/2/5) with protocol version 2 to fd 7
renderd[146]: DEBUG: Connection 0, fd 7 closed, now 2 left
renderd[146]: DEBUG: Connection 0, fd 8 closed, now 1 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 2
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(6), y(3), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 7, number 3
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(7) xml(ajt), z(3), x(1), y(4), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 8, number 4
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(8) xml(ajt), z(3), x(6), y(4), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 10, number 5
renderd[146]: DEBUG: Got incoming connection, fd 11, number 6
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(10) xml(ajt), z(3), x(1), y(2), mime(image/png), options()
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(11) xml(ajt), z(3), x(6), y(2), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/5/5) with protocol version 2 to fd 9
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/6/2) with protocol version 2 to fd 11
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/1/2) with protocol version 2 to fd 10
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/6/4) with protocol version 2 to fd 8
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/1/4) with protocol version 2 to fd 7
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/6/3) with protocol version 2 to fd 6
renderd[146]: DEBUG: Connection 0, fd 9 closed, now 5 left
renderd[146]: DEBUG: Connection 1, fd 7 closed, now 4 left
renderd[146]: DEBUG: Connection 2, fd 10 closed, now 3 left
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 2 left
renderd[146]: DEBUG: Connection 1, fd 11 closed, now 1 left
renderd[146]: DEBUG: Connection 0, fd 8 closed, now 0 left
renderd[146]: DEBUG: Got incoming connection, fd 6, number 1
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(6) xml(ajt), z(3), x(1), y(5), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 7, number 2
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(7) xml(ajt), z(3), x(6), y(5), mime(image/png), options()
renderd[146]: DEBUG: Got incoming connection, fd 8, number 3
renderd[146]: DEBUG: Got incoming request with protocol version 2
renderd[146]: DEBUG: Got command RenderPrio fd(8) xml(ajt), z(3), x(4), y(4), mime(image/png), options()
renderd[146]: Received request for map layer 'ajt' which failed to load
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/1/5) with protocol version 2 to fd 6
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/4/4) with protocol version 2 to fd 8
renderd[146]: DEBUG: Sending render cmd(4 ajt 3/6/5) with protocol version 2 to fd 7
renderd[146]: DEBUG: Connection 0, fd 6 closed, now 2 left
renderd[146]: DEBUG: Connection 1, fd 8 closed, now 1 left
renderd[146]: DEBUG: Connection 0, fd 7 closed, now 0 left
```

I'm guessing this is causing the problems:
> renderd[146]: Received request for map layer 'ajt' which failed to load

but unsure as to what is could be causing this

**Comment by chris442** [comment]
I tried running the container without a volume to preserve rendered tiles and noticed this in the logs after:

```
renderd[145]: An error occurred while loading the map layer 'ajt': Postgis Plugin: FATAL:  role "renderer" does not exist
Connection string: ' dbname=gis connect_timeout=4'
  encountered during parsing of layer 'landcover-low-zoom' in Layer at line 825 of '/home/renderer/src/openstreetmap-carto/mapnik.xml'
renderd[145]: An error occurred while loading the map layer 'ajt': Postgis Plugin: FATAL:  role "renderer" does not exist
Connection string: ' dbname=gis connect_timeout=4'
  encountered during parsing of layer 'landcover-low-zoom' in Layer at line 825 of '/home/renderer/src/openstreetmap-carto/mapnik.xml'
renderd[145]: An error occurred while loading the map layer 'ajt': Postgis Plugin: FATAL:  role "renderer" does not exist
Connection string: ' dbname=gis connect_timeout=4'
  encountered during parsing of layer 'landcover-low-zoom' in Layer at line 825 of '/home/renderer/src/openstreetmap-carto/mapnik.xml'
renderd[145]: An error occurred while loading the map layer 'ajt': Postgis Plugin: FATAL:  role "renderer" does not exist
Connection string: ' dbname=gis connect_timeout=4'
  encountered during parsing of layer 'landcover-low-zoom' in Layer at line 825 of '/home/renderer/src/openstreetmap-carto/mapnik.xml'
```

**Comment by chris442** [comment]
I still uncertain as to what could have caused this but I tried with a new map file (https://download.geofabrik.de/europe/great-britain/scotland.html) and it worked!

The only deviation I made from the setup guide was that I renamed the volume where the tiles are stored in step 1 to `scotlandmap-data`. Somehow I doubt that was the issue, maybe something to do with the original map fiile I tried, or some other weird edge case.
